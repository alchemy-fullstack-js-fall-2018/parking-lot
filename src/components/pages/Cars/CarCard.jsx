import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../../routes';
import styled from 'styled-components';

const StyledDiv =  styled.div`
  margin: 5px;
  border: 1px solid black;
`;

const CarCard = ({ plate, make, model, id }) => {
  return (
    <StyledDiv>
      <p>plate: {plate}</p>
      <p>make: {make}</p>
      <p>model: {model}</p>
      <Link
        to={ROUTES.CAR_DETAIL.linkTo(id)}
      >See more information
      </Link>
    </StyledDiv>
  );
};

CarCard.propTypes = {
  plate: PropTypes.string.isRequired,
  make: PropTypes.string.isRequired,
  model: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default CarCard;
