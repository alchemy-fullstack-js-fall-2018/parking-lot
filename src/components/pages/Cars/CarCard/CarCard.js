import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import styles from './CarCard.css';
import { ROUTES } from '../../../../routes';


const CarCard = ({ plate, make, model, id }) => {
  return (
    <div className={styles.carCard}>
      <p>plate: {plate}</p>
      <p>make: {make}</p>
      <p>model: {model}</p>
      <Link
        to={ROUTES.CAR_DETAIL.linkTo(id)}
      >See more information
      </Link>
    </div>
  );
};

CarCard.propTypes = {
  plate: PropTypes.string.isRequired,
  make: PropTypes.string.isRequired,
  model: PropTypes.string.isRequired
};

export default CarCard;
