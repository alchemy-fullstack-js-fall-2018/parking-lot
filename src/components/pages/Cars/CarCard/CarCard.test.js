import React from 'react';
import { shallow } from 'enzyme';
import CarCard from './CarCard';

describe('<CarCard />', () => {
  test('renders', () => {
    const wrapper = shallow(<CarCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
