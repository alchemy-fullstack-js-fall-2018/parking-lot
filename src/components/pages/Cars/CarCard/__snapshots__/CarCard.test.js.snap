// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarCard /> renders 1`] = `
<div
  className="carCard"
>
  <p>
    plate: 
    abcdef
  </p>
  <p>
    make: 
    Ford
  </p>
  <p>
    model: 
    F150
  </p>
  <Link
    replace={false}
    to="/cars/123-123-123-123"
  >
    See more information
  </Link>
</div>
`;
